### ISSUE TITLE
False positive :unused-excluded-var when shadowed by :refer + :rename

### ISSUE THREAD

**Issue by yuhan0** [clj-kondo/clj-kondo#2813]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**: clj-kondo v2026.04.15

**macro usage**: NA

**platform**: native

**editor**: NA

**problem**

`:refer-clojure :exclude [...]` in an `ns` form followed by a `:rename` from another ns that brings in the excluded symbol is flagged as `:unused-excluded-var`, even though the exclusion is necessary - runtime emits a warning otherwise.

**repro**

```clojure
(ns repro
  (:refer-clojure :exclude [defrecord])
  (:require [potemkin :refer [defrecord+] :rename {defrecord+ defrecord}]))

(defrecord Foo [])
```

```bash
$ clj-kondo --lint repro.clj
repro.clj:2:29: info: Unused excluded var: defrecord
linting took 19ms, errors: 0, warnings: 0
```

**config** NA

**expected behavior**

No output, similar to how the following case (without `:rename`) doesn't produce any:

```clojure
(ns repro2
  (:refer-clojure :exclude [read-string])
  (:require [clojure.edn :refer [read-string]]))
(read-string "1")
```

**Additional info** 

Probably some extra term to be added hereabouts?
https://github.com/clj-kondo/clj-kondo/blob/3b6fbc3079fcfef91ceffe7461eb0570fe13af57/src/clj_kondo/impl/linters.clj#L800-L802

**Comment by borkdude** [comment]
Good edge case. Thanks for reporting.

**Comment by yuhan0** [comment]
Hmm, this made me realise that dropping the `:refer-clojure :exclude` form altogether doesn't result in any lint warnings - is that a false negative or a deliberate choice? i.e.

```clojure
(ns repro3
  (:require [clojure.edn :refer [read-string]]))
(read-string "1")
```
Tried searching for any existing linters/issues but couln't find any

**Comment by borkdude** [comment]
@yuhan0 I think it's an opportunity for a new linter, maybe `:replaced-var` with the message:

> read-string already refers to clojure.core/read-string, being replaced by clojure.edn/read-string
